posts/2025-06-03-25.0.0.6-beta.adoc

```diff
@@ -0,0 +1,187 @@
+
+
+In the previous link:https://blogs-draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/blog/2025/04/08/25.0.0.4-beta.html[25.0.0.4-beta] release, the Open Liberty server introduced the link:https://blogs-draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/blog/2025/04/08/25.0.0.4-beta.html#backport[file-based health check mechanism] for a MicroProfile Health 4.0 feature (`mpHealth-4.0`). In 25.0.0.4-beta, the configuration for this mechanism was introduced with the `server.xml` attribute `fileUpdateInterval` for the `<mpHealth>` element with an associated `MP_HEALTH_FILE_UPDATE_INTERVAL` environment variable. The configuration attribute has now been updated to `checkInterval` with an associated `MP_HEALTH_CHECK_INTERVAL` environment variable. Similar to before, this attribute needs to be configured to enable this alternative file-based health check mechanism. The configuration accepts a number followed by an optional `ms` or `s` to denote the time unit. If no time unit is specified, it defaults to seconds. Invalid values default the interval to 10 s. 
+
+This beta now introduces a new `server.xml` attribute for the `<mpHealth>` element named `startupCheckInterval` with an associated `MP_HEALTH_STARTUP_CHECK_INTERVAL` environment variable. This configuration is optional and defaults to 100 ms if the user has not configured it. The configuration accepts a number followed by an optional `ms` or `s` to denote the time unit. If no time unit is specified, it defaults to milliseconds. Invalid values default the interval to 100 ms.  
+
+The `startupCheckInterval` is used during the _startup_ phase of the file-based health check mechanism. As a refresher, the file-based health check functionality creates the `live`, `ready`, and `started` files under the `${server.output.dir}/health` directory. The `${server.output.dir}` defaults to `wlp/usr/servers/<server>`. For more information, see link:{url-prefix}/docs/latest/reference/directory-locations-properties.html[directory locations and properties]. 
+
+These files are created when their respective health checks return an `UP` status. The `startupCheckInterval` is the interval at which the runtime queries the health checks during the startup phase. When all statuses report `UP` then all three `started`, `ready` and `live` files are created. Next, proceed to the _update_ phase. At this point, `started` file is no longer useful. After Kubernetes has found an `UP` status, it proceeds to start its checks for _liveness_ and _readiness_. It's at this point when the `checkInterval` is used by the runtime to establish the interval at which the liveness and readiness health checks are queried. Similar to before, if an `UP` status is resolved then the respective health-check file is _updated_ (that is, the last modified time of the file is updated). A `DOWN` status results in the respective health-check file not being updated.
```

Comment:
> Next, proceed to the _update_ phase.

Who or what is proceeding to that phase? Me? The application? The server? I don't understand what's meant by this here.

> At this point, `started` file is no longer useful.
I'd suggest rephrasing this to say why that file is no longer useful. As I understand it, the file is only useful as a flag to indicate that all applications with health checks have started. So yes, after they've all started, I can understand the file does not server much purpose anymore. But I'd suggest at least adding a phrase here to point out why.